instrumentation/external-annotations/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/extannotations/TraceProvidersTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.extannotations;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.semconv.SemanticAttributes;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Locale;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/** This test verifies that Otel supports various 3rd-party trace annotations */
+class TraceProvidersTest {
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @ParameterizedTest
+  @MethodSource("provideArguments")
+  void testShouldSupportProvider(String provider)
+      throws ClassNotFoundException,
+          NoSuchMethodException,
+          InstantiationException,
+          IllegalAccessException,
+          InvocationTargetException {
+
+    Class<?> cls =
+        Class.forName("io.opentelemetry.javaagent.instrumentation.extannotations.SayTracedHello");
```

Comment:
I think you don't need the `Class.forName` here, could directly use `SayTraceHello.class`. Or you could do something like
```java
  @ParameterizedTest
  @EnumSource(TraceProvider.class)
  void testShouldSupportProvider(TraceProvider provider) {
    provider.test();

    testing.waitAndAssertTraces(
        trace ->
            trace.hasSpansSatisfyingExactly(
                span ->
                    span.hasName("SayTracedHello." + provider.testMethodName())
                        .hasAttributesSatisfyingExactly(
                            equalTo(
                                SemanticAttributes.CODE_NAMESPACE, SayTracedHello.class.getName()),
                            equalTo(SemanticAttributes.CODE_FUNCTION, provider.testMethodName()),
                            equalTo(stringKey("providerAttr"), provider.name()))));
  }

  @SuppressWarnings("ImmutableEnumChecker")
  private enum TraceProvider {
    AppOptics((sayTracedHello) -> sayTracedHello.appoptics()),
    Datadog((sayTracedHello) -> sayTracedHello.datadog()),
    Dropwizard((sayTracedHello) -> sayTracedHello.dropwizard()),
    KamonOld((sayTracedHello) -> sayTracedHello.kamonold()),
    KamonNew((sayTracedHello) -> sayTracedHello.kamonnew()),
    NewRelic((sayTracedHello) -> sayTracedHello.newrelic()),
    SignalFx((sayTracedHello) -> sayTracedHello.signalfx()),
    Sleuth((sayTracedHello) -> sayTracedHello.sleuth()),
    Tracelytics((sayTracedHello) -> sayTracedHello.tracelytics());

    final Consumer<SayTracedHello> test;

    TraceProvider(Consumer<SayTracedHello> test) {
      this.test = test;
    }

    void test() {
      test.accept(new SayTracedHello());
    }

    String testMethodName() {
      return name().toLowerCase(Locale.ROOT);
    }
  }
```

Reply:
🤯 that is a much more elegant approach, thank you!